src/node/executor.cpp

```diff
@@ -54,8 +54,11 @@ void Executor::stop() {
 }
 
 void Executor::execute(std::shared_ptr<PbftBlock> blk) {
-  std::unique_lock l(mu_);
-  to_execute_ = move(blk);
+  assert(final_chain_->last_block_number() < blk->getPeriod());
+  {
```

Comment:
Added brackets to exclude `notify_one` from the critical section, because it doesn't have to be done under the lock, and in fact it would be a pessimization